src/blueapi/service/handler.py

```diff
@@ -14,37 +14,46 @@ class Handler:
     context: BlueskyContext
     worker: Worker
     config: ApplicationConfig
-    message_bus: MessagingTemplate
-
-    def __init__(self, config: Optional[ApplicationConfig] = None) -> None:
-        self.context = BlueskyContext()
-        self.config = config if config is not None else ApplicationConfig()
+    messaging_template: MessagingTemplate
+
+    def __init__(
+        self,
+        config: Optional[ApplicationConfig] = None,
+        context: Optional[BlueskyContext] = None,
+        messaging_template: Optional[MessagingTemplate] = None,
+        worker: Optional[Worker] = None,
+    ) -> None:
+        self.config = config or ApplicationConfig()
+        self.context = context or BlueskyContext()
 
         logging.basicConfig(level=self.config.logging.level)
 
         self.context.with_config(self.config.env)
 
-        self.worker = RunEngineWorker(self.context)
-        self.message_bus = StompMessagingTemplate.autoconfigured(self.config.stomp)
+        self.worker = worker or RunEngineWorker(self.context)
+        self.messaging_template = (
+            messaging_template
+            or StompMessagingTemplate.autoconfigured(self.config.stomp)
+        )
 
     def start(self) -> None:
         self.worker.start()
 
         self._publish_event_streams(
             {
-                self.worker.worker_events: self.message_bus.destinations.topic(
+                self.worker.worker_events: self.messaging_template.destinations.topic(
                     "public.worker.event"
                 ),
-                self.worker.progress_events: self.message_bus.destinations.topic(
+                self.worker.progress_events: self.messaging_template.destinations.topic(
                     "public.worker.event.progress"
                 ),
-                self.worker.data_events: self.message_bus.destinations.topic(
+                self.worker.data_events: self.messaging_template.destinations.topic(
                     "public.worker.event.data"
                 ),
             }
         )
 
-        self.message_bus.connect()
+        self.messaging_template.connect()
```

Comment:
why the change to `messaging_template`?

I think I initially changed this from `template` to `message_bus` because I thought it's easier to understand. I know its perhaps a bit silly because, the `message_bus` object isn't actually a message bus, it's a connection to a message bus. However... I'm not sure I like messaging template all that much, or the word 'template'. 

But happy for that to be another discussion in another issue/PR because we have a few things named 'template'.